spec/controllers/auth_controller_spec.rb

```diff
@@ -0,0 +1,134 @@
+                            'email' => 'hi@natashatherobot.com',
+                            'nickname' => 'NatashaTheRobot'
+                          },
+                          'credentials' =>
+                            {
+                              'token' => '445da8f3214dafaaffb11f9f71c2ed8612287ac8',
+                              'expires' => false
+                            },
+
+                          'extra' =>
+                            {
+                              'raw_info' => {
+                                'login' => 'natasha'
+                              }
+                            }
+                        } }
+    before(:each) do
+      request.env['omniauth.auth'] = omniauth_hash
+    end
+
+    context "user not logged in" do
+      it "should create a new user if it does not already exist" do
+        get :success, provider: 'github'
+        expect(User.find_by_email(omniauth_hash['info']['email'])).to_not be_nil
+      end
+
+      it "authenticates the user" do
+        get :success, provider: 'github'
+        expect(request.env['warden'].user.uid).to eq(omniauth_hash['uid'])
+      end
+    end
+
+    context 'user has no websites' do
+      it "redirects to website_wizard_path#create if the user has no websites" do
+        no_websites_user = FactoryGirl.create(:user, { provider: omniauth_hash['provider'], uid: omniauth_hash['uid'], name: omniauth_hash['info']['name'], email: omniauth_hash['info']['email'] })
+        get :success, provider: 'github'
+        expect(response).to redirect_to(website_wizard_path(:create))
+      end
+    end
+
+    context 'user exists and has websites' do
+      let!(:new_user) { create :user, provider: omniauth_hash['provider'], uid: omniauth_hash['uid'], name: omniauth_hash['info']['name'], email: omniauth_hash['info']['email'] }
+      let!(:new_website_member) { create :website_member, user: new_user, website: website }
+
+      it "sets the website to the first website of the user" do
+        get :success, provider: 'github'
+        expect(session['epiclogger_website_id']).to eq(new_user.websites.first.id)
+      end
+
+      it "redirects to errors_url if a website was selected" do
+        get :success, provider: 'github'
+        expect(response).to redirect_to(errors_path())
+      end
+
+      context "with integration_session" do
+        it "creates a new integration based on the auth hash" do
+          session[:integration] = { "name" => "Github", "provider" => "github" }
+          get :success, provider: 'github'
+          expect(assigns(:integration).website).to eq(website)
+        end
+
+        it "redirects to the installations path" do
+          session[:integration] = { "name" => "Github", "provider" => "github" }
+          get :success, provider: 'github'
+          expect(response).to redirect_to(installations_path(main_tab: 'integrations', integration_tab: omniauth_hash['provider'] ))
```

Comment:
See that we removed the installations path so you'll need to change these to settings_path